traitsui/tests/editors/test_tabular_editor.py

```diff
@@ -39,30 +45,326 @@ class Report(HasTraits):
-                selected="selected",
-                selected_row="selected_row",
-                refresh="refresh",
-                update="update",
-            ),
+
+def get_view(multi_select=False):
+    if multi_select:
+        return View(
+            Item(
+                name="people",
+                editor=TabularEditor(
+                    adapter=ReportAdapter(),
+                    selected="multi_selected",
+                    selected_row="selected_rows",
+                    refresh="refresh",
+                    update="update",
+                    multi_select=True,
+                ),
+            )
+        )
+    else:
+        return View(
+            Item(
+                name="people",
+                editor=TabularEditor(
+                    adapter=ReportAdapter(),
+                    selected="selected",
+                    selected_row="selected_row",
+                    refresh="refresh",
+                    update="update",
+                ),
+            )
+        )
+
+
+def get_selected_rows(editor):
+    """ Returns a list of all currently selected rows.
+    """
+    if is_current_backend_wx():
+        import wx
+        # "item" in this context means "row number"
+        item = -1
+        selected = []
+        while True:
+            item = editor.control.GetNextItem(
+                item, wx.LIST_NEXT_ALL, wx.LIST_STATE_SELECTED
+            )
+            if item == -1:
+                break
+            selected.append(item)
+        return selected
+
+    elif is_current_backend_qt4():
+        rows = editor.control.selectionModel().selectedRows()
+        return [r.row() for r in rows]
+
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+def set_selected_single(editor, row):
+    """ Selects a specified row in an editor with multi_select=False.
+    """
+    if is_current_backend_wx():
+        editor.control.Select(row)
+
+    elif is_current_backend_qt4():
+        from pyface.qt.QtGui import QItemSelectionModel
+
+        smodel = editor.control.selectionModel()
+        mi = editor.model.index(row, 0)
+        smodel.select(
+            mi, QItemSelectionModel.ClearAndSelect | QItemSelectionModel.Rows
         )
-    )
 
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+def set_selected_multiple(editor, rows):
+    """ Clears old selection and selects specified rows in an editor
+    with multi_select=True.
+    """
+    if is_current_backend_wx():
+        clear_selection(editor)
+        for row in rows:
+            editor.control.Select(row)
+
+    elif is_current_backend_qt4():
+        from pyface.qt.QtGui import QItemSelectionModel
+
+        clear_selection(editor)
+        smodel = editor.control.selectionModel()
+        for row in rows:
+            mi = editor.model.index(row, 0)
+            smodel.select(
+                mi, QItemSelectionModel.Select | QItemSelectionModel.Rows
```

Comment:
Same here, it is not obvious why the `Rows` flag is needed. A comment will be very helpful.